x-pack/legacy/plugins/canvas/canvas_plugin_src/renderers/time_filter/components/time_picker/time_picker.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import dateMath from '@elastic/datemath';
+import { EuiButton } from '@elastic/eui';
+import moment from 'moment';
+import { DatetimeQuickList } from '../datetime_quick_list';
+import { DatetimeRangeAbsolute } from '../datetime_range_absolute';
+
+export interface Props {
+  /** Start date string */
+  from: string;
+  /** End date string */
+  to: string;
+  /** Function invoked when date range is changed */
+  onSelect: (from: string, to: string) => void;
+}
+
+export interface State {
+  range: {
+    from: string;
+    to: string;
+  };
+  isDirty: boolean;
+}
+
+export class TimePicker extends Component<Props, State> {
+  static propTypes = {
+    from: PropTypes.string.isRequired,
+    to: PropTypes.string.isRequired,
+    onSelect: PropTypes.func.isRequired,
+  };
+
+  state = {
+    range: { from: this.props.from, to: this.props.to },
+    isDirty: false,
+  };
+
+  componentWillReceiveProps({ from, to }: Props) {
```

Comment:
Isn't `componentWillReceiveProps` getting deprecated? It's used in `datetime_input/index.ts` as well. You don't need to address this in this PR, but it'd be nice if you could add a TODO here to refactor this later.

Reply:
yup! good call